straxbra/xebra_context.py

```diff
@@ -38,3 +41,57 @@ def get_array(self, run_id, *args, **kwargs) -> np.ndarray:
     def get_df(self, run_id, *args, **kwargs) -> pd.DataFrame:
         run_id = process_runlist(run_id)
         return super().get_df(run_id, *args, **kwargs)
+
+
+
+@export
+class HtpcContext(strax.Context):
+
+    std_dtypes = ('raw_records', 'records', 'peaks', 'events')
+    # new plugins must be listed here in order to be registered
+    plugins_to_register = 'DAQReader Records Peaks PeakBasics PeakClassification ' \
+                          'NCompeting Events EventHermBasics EventHermPositions'.split(' ')
+
+    def __init__(self, *args, **kwargs):
+        experiment = kwargs.pop('experiment', 'htpc')
+
+        if 'config' not in kwargs:
+            kwargs['config'] = {'experiment': experiment}
+        elif 'experiment' not in kwargs['config']:
+            kwargs['config']['experiment'] = experiment
+        if 'storage' not in kwargs:
+            kwargs['storage'] = os.path.join(storage_base_dir, experiment)
+
+        # add configs that must be (/you wanto to be) different from xebra dual phase tpc here
+        # n_channels and drift_length/drift_vel is set in utils.py - relies on run_id starting with '1'
+        configs = {                  # type       plugin       xebra_val
+                'hit_threshold': 7,  #  int   records,peaks       30
+                'top_pmts': [1]      # list   p_basics,p_pos    list(range(1,8))
+                }
+
+        for config_name, config_value in configs.items():
+            if config_name not in kwargs['config']:
+                kwargs['config'][c_name] = c_value
+        
+        if 'register' not in kwargs and 'register_all' not in kwargs:
+            register = []
+            # mostly stolen from strax's context.py
```

Comment:
I recommend against stealing code in this manner

Reply:
That's actually not the code I stole. What I stole is directly underneath the comment:

```
for plugin in dir(plugins):
                plugin = getattr(plugins, plugin)
                if type(plugin) != type(type):
                    continue
                if (
                        issubclass(plugin, strax.Plugin) and
                        plugin.__name__ in self.plugins_to_register):

                    register.append(plugin)
            kwargs['register'] = register
```



I did this because I didn't want to register all the plugins in plugins.py, so I check all the plugins if their name is in the ones I want to register. The code is in 'register_all' that uses the code to pass all single plugins to the 'register' method.

I also adopted the syntax that you used in the code you marked here. I guess there is a nicer way to do it, you're right. I'll suggest a change in the next commit.

Reply:
Making copies of plugins and hard-borrowing code is probably much more work than just making a neural net that always returns (0,0) for the (x,y) coordinates, which heads off most of the issues at the pass.

Reply:
I see. That sounds better, in deed. So I'd make the NN-model a strax-option and create a model as you suggested.

Passing the path to the model as a strax option is probably not best, since the same model can have different names (hash changes while model doesn't) and you can edit the model and give it the same name (same hash, different models). Should we leave that to the user or maybe not even track it as it's currently, effectively done since its hard coded and not a strax option?

And then just register_all?

Reply:
If you have the same model with different filenames then you've dug your own hash-based grave. Not tracking this is also not a good idea, because then when you change the model because <insert reason here> then strax won't have any way of knowing, so it'll just load the stuff that's already on disk.

Reply:
As discussed I just set the `min_reconstruction_area` to 10¹⁰ PE which effectively skips the reconstruction of every peak. That also comes with the next commit.

So I can still make the path to the model a strax-option and let it be tracked. Meanwhile I should give the htpc its own fake model, so that the htpc-hash is not changed when the xebra model changes. Do you want me to do that?

Reply:
It's probably better to have a separate model for the hermetic detector.

Reply:
Done.
I'll create a repo for the nn-models, too.